components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/ContentMarshaller.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.view;
+
+import java.util.Date;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import org.sonatype.nexus.common.collect.AttributesMap;
+import org.sonatype.nexus.common.collect.NestedAttributesMap;
+import org.sonatype.nexus.common.hash.HashAlgorithm;
+import org.sonatype.nexus.repository.proxy.CacheInfo;
+import org.sonatype.nexus.repository.proxy.CacheInfoMarshaller;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+
+import com.google.common.collect.Maps;
+import com.google.common.hash.HashCode;
+import org.joda.time.DateTime;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.common.time.DateHelper.toDate;
+import static org.sonatype.nexus.common.time.DateHelper.toDateTime;
+
+/**
+ * Extracts non-format specific auxiliary metadata from asset.
+ *
+ * @since 3.0
+ */
+public class ContentMarshaller
```

Comment:
@cstamas don't mind if you merge them or not (just pointing out that might be convenient to do so rather than have related methods sitting in a separate class) but the name definitely needs sorting out here as this doesn't marshal content, it only marshals content metadata.